appservice/src/createAppService/SiteRuntimeStep.ts

```diff
@@ -75,23 +63,90 @@ export class SiteRuntimeStep extends AzureWizardPromptStep<IAppServiceWizardCont
+            if (aInfo.name !== bInfo.name) {
+                const result: number = getPriority(aInfo.name) - getPriority(bInfo.name);
+                return result === 0 ? a.displayVersion.localeCompare(b.displayVersion) : result;
+            } else if (aInfo.major !== bInfo.major) {
+                return bInfo.major - aInfo.major;
+            } else {
+                return bInfo.minor - aInfo.minor;
+            }
+        })
+        // convert to quick pick
+        .map(mv => {
+            return {
+                id: mv.runtimeVersion,
+                label: mv.displayVersion,
+                data: mv.runtimeVersion,
+                // include stack as description if it has a version
+                // tslint:disable-next-line: strict-boolean-expressions
+                description: /[0-9]/.test(mv.stackDisplay) ? mv.stackDisplay : undefined
+            };
+        });
+}
 
-        const runtimesParsed: ApplicationStackJsonResponse = <ApplicationStackJsonResponse>JSON.parse(runtimes);
+interface IParsedRuntimeVersion {
+    name: string;
+    major: number;
+    minor: number;
+}
 
-        return runtimesParsed.value.map((runtime) => {
-            return nonNullProp(runtime.properties, 'majorVersions').map((majorVersion) => {
-                return { name: majorVersion.runtimeVersion, display: majorVersion.displayVersion, isDefault: majorVersion.isDefault };
-            });
-        }).reduce((acc, val) => acc.concat(val));
-        // this is to flatten the runtimes to one array
+function getRuntimeInfo(runtimeVersion: string): IParsedRuntimeVersion {
+    const parts: string[] = runtimeVersion.split('|');
+
+    let major: string | undefined;
+    let minor: string | undefined;
+    if (parts[1]) {
+        const match: RegExpMatchArray | null = parts[1].match(/([0-9]+)(?:\.([0-9]+))?/);
+        if (match) {
+            major = match[1];
+            minor = match[2];
+        }
     }
 
-    private sortQuickPicksByRuntime(runtimeItems: IAzureQuickPickItem<ApplicationStack>[], recommendedRuntimes: string[]): IAzureQuickPickItem<ApplicationStack>[] {
-        function getPriority(item: IAzureQuickPickItem<ApplicationStack>): number {
+    return {
+        name: parts[0],
+        major: convertToNumber(major),
+        minor: convertToNumber(minor)
+    };
+}
 
-            const index: number = recommendedRuntimes.findIndex((runtime: string) => nonNullProp(item.data, 'name').includes(runtime));
-            return index === -1 ? recommendedRuntimes.length : index;
+function convertToNumber(data: string | undefined): number {
+    if (data) {
+        const result: number = parseInt(data);
+        if (!isNaN(result)) {
+            return result;
         }
-        return runtimeItems.sort((a: IAzureQuickPickItem<ApplicationStack>, b: IAzureQuickPickItem<ApplicationStack>) => getPriority(a) - getPriority(b));
     }
+    return 0;
```

Comment:
Rather than return 0, I feel like we should return a higher number so that LTS gets displayed at the top.  I think it's what the Portal is trying to lead people to at least.

Reply:
Fine with me